### DISCUSSION TITLE
Fluent mapping

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [linq2db/linq2db#3180] [General]
I saw a lot of people using fluent mappping. I am new to linq2db,, just want to be sure of somethings:

If I use t4template, attribute baseed mapping, then all fluentmapping solutions is invalid, because the schema mappings are readonly.

Or maybe I am wrong? we cans till use fluentmapping to modify or add additional mapping info to the base mapping?

Thanks

**Comment by MaceWindu** [comment]
You can combine mapping sources but it is not recommended as you need to understand how it works to configure such cases properly.
You will have two metadata providers that could return metadata (e.g. `TableAttribute` or `ColumnAttribute`) and if both of them return specific piece of metadata (e.g. `ColumnAttribute` for specific entity property) - only first one will be used and others will be ignored.

Which means:
1. you cannot modify only one property of metadata attribute - you need to specificy all data provided by this attribute (at least data that is important for your data model)
2. your fluent mapping metadata provider should be queried first for metadata to take priority. Fortunatelly additional metadata providers added with higher priority than already registered providers so it should work for you without additional effort

Also, another option to give fluent metadata higher priority is to specify `Configuration` name (T4 generates attributes without explicit `Configuration` so they will have lower priority), but currently it's poorly supported by fluent mapper (see https://github.com/linq2db/linq2db/issues/3136)
